wordpress-mobile/WordPress-Android
# [Modules Restructuring] Restructure Image Editor Module

Parent #16562
Fixes #16565

This PR restructures the `image-editor` module removing the extra `ImageEditor` level from it. This effectively makes `image-editor` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

PS.1: @ashiagr I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since you helped us with the [navigation component](https://github.com/wordpress-mobile/WordPress-Android/pull/16632) update recently and its testing. As such, this should be very straightforward for you to test again, verify correctness and sign-off on that change for WPAndroid. 🥇 

PS.2: @AliSoftware I added you as a secondary reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, as I did a [localization.rb](https://github.com/wordpress-mobile/WordPress-Android/pull/16765/commits/90d6bd1643b8dc53183a765297587982635afd81) related change that I want you to double check and verify against, just in case I missed something vital. 🥇 

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I don't see any `LICENCE` file for this module, like I did for the `analytics` mode, which is pointing to `MIT`, I am not sure if we should include a licence for this module too or exclude the licensing information from the `analytics` and all other modules in general so that we start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2`, wdyt? 🤔 

-----

To test:
- Build the app and install it on your device or emulator.
- Go to `Posts` -> Click on the plus 🔵 FAB button on the bottom right -> Select `Blog post` to create a new post.
- Search for and add an `Image` block -> Choose an image from either of the options available. -> Use this photo.
- Click the `Media options` cycle button on the top right of this photo -> Select `Edit`.
- Crop this photo and/or do any other editing -> Then click on the ✓ icon on the top right.
- Verify that the photo editing worked and that the photo is updated accordingly.

The above is just one example of how this change can tested so that we could quickly verify the overall correctness of this PR. However, feel free to smoke test other aspects as well, especially if you are aware any or all the other places where `image-editor` is being used.

PS: In addition to the above, you might also want to verify that the `image-editor` testing command works as expect by just copy-pasting this command below and running it locally. This is because I did a Buildkite [pipeline.yml](https://github.com/wordpress-mobile/WordPress-Android/pull/16765/commits/22ec034232853380c2ae447b9f8bd79338aebdcb) related change as part of this PR as well.

`./gradlew :libs:image-editor:test`

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
